src/components/internals/wrapper-with-icon/wrapper-with-icon.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { css } from '@emotion/core';
+import { customProperties } from '@commercetools-uikit/design-system';
+import { components as defaultComponents } from 'react-select';
+
+const getDefaultComponent = (type) => {
+  if (type === 'singleValue') return defaultComponents.SingleValue;
+  if (type === 'placeholder') return defaultComponents.Placeholder;
+  return React.Fragment;
+};
+
+const WrapperWithIcon = ({ children, ...props }) => {
+  const Component = getDefaultComponent(props.type);
+
+  return (
+    <React.Fragment>
+      {props.selectProps.iconLeft &&
+        React.cloneElement(props.selectProps.iconLeft, {
+          size: 'big',
+        })}
+      <span
+        css={css`
+          margin: ${customProperties.spacingXl + customProperties.spacingXs};
+        `}
+      >
+        <Component {...props}>{children}</Component>
+      </span>
+    </React.Fragment>
+  );
+};
+WrapperWithIcon.propTypes = {
+  children: PropTypes.node,
+  type: PropTypes.oneOf(['singleValue', 'placeholder']),
+  selectProps: PropTypes.shape({
+    iconLeft: PropTypes.node,
+  }),
+};
+WrapperWithIcon.displayName = 'WrapperWithIcon';
+
+export default WrapperWithIcon;
+
+/* eslint-disable react/display-name */
+const customComponents = {
+  SingleValue: (props) => <WrapperWithIcon {...props} type="singleValue" />,
+  Placeholder: (props) => <WrapperWithIcon {...props} type="placeholder" />,
```

Comment:
Is this here just incase we show an icon along side with placeholder?

Reply:
Yes, the icon should be shown even before selecting an option, and so it should appear on the placeholder too.